components/cdc/src/endpoint.rs

```diff
@@ -1182,11 +1214,18 @@ mod tests {
             err: Some(Error::Request(err_header.clone())),
         };
         ep.run(Task::Deregister(deregister));
-        let (_, mut change_data_event) = rx.recv_timeout(Duration::from_millis(500)).unwrap();
-        let event = change_data_event.event.take().unwrap();
-        match event {
-            Event_oneof_event::Error(err) => assert!(err.has_not_leader()),
-            _ => panic!("unknown event"),
+        let cdc_event = rx.recv_timeout(Duration::from_millis(500)).unwrap();
+        loop {
```

Comment:
I guess you mean:
```suggestion
        loop {
            let cdc_event = rx.recv_timeout(Duration::from_millis(500)).unwrap();
```

Also, what about setting a large min_ts_interval to avoid resolved ts events?